src/components/a-choice/_template.js

```diff
@@ -7,15 +7,16 @@ export default function ({
   name,
   checked = false,
   disabled = false,
+  controlled = false,
 }, childrenFragment) {
   return html`<label class="a-choice__wrapper">
     <input
       id=${inputId}
-      class="a-choice__input" 
+      class="a-choice__input a-choice__input--${controlled ? 'controlled' : 'uncontrolled'}"
```

Comment:
This is indeed a bug.

Though I would not fix the symptom by adding overruling classes.
Rather I assume it to be broken at the our DOM-diffing level.